### PR TITLE
Add clarification to doc on cluster cross boundary use of DeathWatch

### PR DESCRIPTION
**Pull request by helena**
#27376

### CODE REVIEW

**Comment by helena** [comment]
Travis CI reporting issue but it passed https://travis-ci.org/akka/akka/builds/561070335

**Comment by helena** [comment]
Running a multi-jvm nightly against it now.

**Comment by helena** [comment]
cc @skvithalani if you'd like to add any thoughts.

**Comment by skvithalani** [comment]
> cc @skvithalani if you'd like to add any thoughts.

I am ok with the clarifications. But this is how I am explaining this to our team:

During death-watch of an actorRef, it is possible that `watcher` and `watchee` do not reside in the same cluster. This could happen in one of the following 3 scenarios:
- one of them is in cluster-A and the other one is in cluster-B
- one of them is in cluster-A and the other one is using a remote provider
- both of them are using a remote provider

In all these 3 scenarios, death-watch will fail in 2.6 unless we enable `use-unsafe-remote-features-outside-cluster= on` on the `watcher` side. 

The last point is key. The flag must be on the watcher side in all 3 scenarios whether or not `watcher` is a cluster.

**Comment by helena** [comment]
@skvithalani Also note that if in the same cluster, it will not work until the watcher/watchee has joined the cluster, noted in the docs.

File: akka-docs/src/main/paradox/project/migration-guide-2.5.x-2.6.x.md [hunk 1]
```diff
@@ -271,15 +271,20 @@ By default, these remoting features are disabled when not using Akka Cluster:
 * Remote Deployment: falls back to creating a local actor
 * Remote Watch: ignores the watch and unwatch request, and `Terminated` will not be delivered when the remote actor is stopped or if a remote node crashes
 
-When used with Cluster, all previous behavior is the same except a remote watch of an actor is no longer possible before a node joins a cluster, only after.
+When used within a Cluster, all previous behavior is the same except a remote watch of an actor is no longer possible before a node joins a cluster, only after.
```
Comment (jrudolph) [inline akka-docs/src/main/paradox/project/migration-guide-2.5.x-2.6.x.md]: Here's a bit of a suggestion for a restructuring of that paragraph:

```
Watching an actor on a node outside the cluster may have unexpected
@ref[consequences](../remoting-artery.md#quarantine), such as quarantining
so it has been disabled by default in Akka 2.6. This is either the case if
cluster is not used at all (but only plain remoting) or when watching an actor outside of the cluster.

On the other hand, failure detection between nodes, that are members of the same cluster,
do not have that shortcoming. Thus, when remote watching or deployment is used within
a cluster, they are working the same in 2.6 as before except that a remote watch of
an actor will be ignored (?correct?) before a node has joined a cluster and works only after.

To optionally enable...
```

**Review by jrudolph — COMMENTED** [review]
LGTM, suggestion for slightly clarifying the changes and rationale below.

File: akka-remote/src/main/resources/reference.conf [hunk 2]
```diff
@@ -166,10 +166,10 @@ akka {
 
     # If Cluster is not used, remote watch and deployment are disabled.
     # To optionally use them while not using Cluster, set to 'on'.
-    use-unsafe-remote-features-without-cluster = off
+    use-unsafe-remote-features-outside-cluster = off
```
Comment (raboof) [inline akka-remote/src/main/resources/reference.conf]: :+1: this seems clearer and more accurate

File: akka-remote/src/main/resources/reference.conf [hunk 3]
```diff
@@ -166,10 +166,10 @@ akka {
...
 
     # A warning will be logged on remote watch attempts if Cluster
-    # is not in use and 'use-unsafe-remote-features-without-cluster'
+    # is not in use and 'use-unsafe-remote-features-outside-cluster'
     # is 'off'. Set this to 'off' to suppress these.
     warn-unsafe-watch-without-cluster = on
```
Comment (raboof) [inline akka-remote/src/main/resources/reference.conf]: Should we also rename this one?

**Review by raboof — COMMENTED** [review]
:+1:, should `warn-unsafe-watch-without-cluster` also be renamed? (could perhaps be a separate issue)

File: akka-docs/src/main/paradox/project/migration-guide-2.5.x-2.6.x.md (hunk 1 above)
Comment (helena) [reply akka-docs/src/main/paradox/project/migration-guide-2.5.x-2.6.x.md]: Great, updated with slight rewording.

File: akka-remote/src/main/resources/reference.conf (hunk 3 above)
Comment (helena) [reply akka-remote/src/main/resources/reference.conf]: good one ;)

**Review by jrudolph — APPROVED** [review]
LGTM

**Review by raboof — APPROVED** [review]

**Pull request merged**